packages/react-jss/tests/defaultPropsWithStyles.tsx

```diff
@@ -0,0 +1,24 @@
+import React, {Component} from 'react'
```

Comment:
I wonder if we should keep all ts tests in one file, otherwise there will be a mixture of js and tsx tests in that folder and that's weird because tsx tests are not jest tests like in those js files.

I think ideally we should move everything typescript into some dir in each package and have there both interfaces and tests

Reply:
I realize this is not really part of the scope here, but with adding this file name it contributes to confusion

Reply:
> I wonder if we should keep all ts tests in one file, otherwise there will be a mixture of js and tsx tests in that folder and that's weird because tsx tests are not jest tests like in those js files.
> 
> I think ideally we should move everything typescript into some dir in each package and have there both interfaces and tests


Oh, thats right. I can do a subfolder `types` just by example of https://github.com/cssinjs/jss/blob/master/packages/jss/tests/types/Styles.ts
Also If you want  I can change file name by your proposal.

Reply:
what about moving everything typescript into {packageName}/ts/ ... and inside for example
`packages/jss/ts/index.d.ts`, `packages/jss/ts/index.test.d.ts` or if tests are specific to some interface `packages/jss/ts/styles.test.ts` etc

Reply:
Hmm, I think that such folder reorganising doesn't make sense. 
For me as for new contributor it is much clearer if all tests will be placed at `tests` folder including `ts` tests.

Reply:
Those ts tests are not the same kind of tests, in fact they are not tests, they are just code that uses types that we use because the code and tests itself is written without ts so nothing is using ts types.

Reply:
Also have you noticed that tests dir is only used for some type of tests, all unit tests are directly located next to the code e.g. JssProvider.js, JssProvider.test.js

Reply:
> Also have you noticed that tests dir is only used for some type of tests, all unit tests are directly located next to the code e.g. JssProvider.js, JssProvider.test.js

Oh, I didn't noticed that. But, follow that logic, I should put my test near the withStyles.js file and call it withStyles.test.tsx.

Reply:
that is exactly the logic I am following, but the thing is this "tests" is testing the interface, so I consider the interfaces file to be "code" and that code using that interface - test

Reply:
Nice, think I understood your point. So I will put my test near the `index.d.ts` and call it `index.test.tsx`. Are you ok with that?

Reply:
yeah, for now that would be ok unless you want to move all ts things into {packageName}/ts

Reply:
Done. I think it's not good idea move index.d.ts to another folder, because it should be placed right near index.js file. Thats how typescript will associate interfaces with implementation.